src/main/java/com/puppycrawl/tools/checkstyle/utils/CommonUtil.java

```diff
@@ -494,45 +497,86 @@ public static void close(Closeable closeable) {
     /**
      * Resolve the specified filename to a URI.
      *
-     * @param filename name os the file
+     * @param filename name of the file
      * @return resolved header file URI
      * @throws CheckstyleException on failure
      */
     public static URI getUriByFilename(String filename) throws CheckstyleException {
-        // figure out if this is a File or a URL
+        URI uri = getWebOrFilePrefixUri(filename);
+
+        if (uri == null) {
+            uri = getClasspathOrFilepathUri(filename);
+        }
+
+        return uri;
+    }
+
+    /**
+     * Resolves the specified filename containing 'http', 'https', 'ftp',
+     * and 'file' protocols (or any RFC 2396 compliant URL) to a URI.
+     *
+     * @param filename name of the file
+     * @return resolved header file URI or null if URL is malformed or non-existent
```

Comment:
> header file 

This is a common utility. It can be used by more than just header files.

Reply:
Done.